Assignment2/Database.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * Establish database connection and perform insertion.
+ */
+class Database {
+
+  // Instance variables to store credentials.
+  private $server_name;
+  private $user_name;
+  private $password;
+  private $db_name;
+
+  // Connection object.
+  public $conn;
+
+  // Variables for holding queries.
+  public $q1;
+  public $q2;
+  public $q3;
+
+  /**
+   * Constructor to initialize credentials and establish connection.
+   *
+   * @param string $server
+   *   Server name.
+   * @param string $user
+   *   Username.
+   * @param string $pwd
+   *   User password.
+   * @param string $db
+   *   Database name to connect to.
+   */
+  function __construct(string $server, string $user, string $pwd, string $db) {
+    $this->server_name = $server;
+    $this->user_name = $user;
+    $this->password = $pwd;
+    $this->db_name = $db;
+    try {
+      $this->conn = new mysqli($this->server_name,$this->user_name,$this->password,$this->db_name);
+    }
+    catch (Exception $e) {
+      die('Connection Failed' . $e);
+    }
+  }
+
+  /**
+   * Function to insert form data to database.
+   *
+   * @return bool
+   *   Returns TRUE on data insertion success and false otherwise.
+   */
+  function insertToData() {
+    $this->q1 = "INSERT INTO employee_code_table (
+      employee_code, employee_code_name, employee_domain) values (
+        '{$_POST["employee_code"]}',
+        '{$_POST["employee_code_name"]}',
+        '{$_POST["employee_domain"]}'
+      )";
+
+    $this->q2 = "INSERT INTO employee_salary_table (
```

Comment:
Can we improve the reusability of this code? Currently, we directly access the required values and perform actions. Instead, we should pass the table name and the data to be inserted as parameters and perform the action, so that it can be reused.

Reply:
There is only one table and even if we do so the column names will not be the same so it won't be reusable in this case. Please let me know in that case if that is correct or not.